### PR TITLE
[UDzPn3dS] Update readme about loading and unrestricting

### PR DESCRIPTION
**Pull request by Lojjs**
Same content changes as for APOC docs: https://github.com/neo4j/docs-apoc/pull/86

### CODE REVIEW

File: readme.adoc [hunk 1]
```diff
@@ -172,13 +172,19 @@ You can find https://github.com/neo4j/apoc/releases/[all releases here].
 [WARNING]
 ====
 // tag::warnings[]
+APOC contains over 450 functions and procedures.
+It is *not* recommended to load all of these into the dbms, but instead use the principle of least privilege.
+This principle dictates, that you should only load the functions and procedures that you strictly need to execute your queries.
```
Comment (JPryce-Aklundh) [inline readme.adoc:177]:
```suggestion
This principle dictates that you only load the functions and procedures necessary to execute your queries.
```

File: readme.adoc [hunk 2]
```diff
@@ -172,13 +172,19 @@ You can find https://github.com/neo4j/apoc/releases/[all releases here].
 [WARNING]
 ====
 // tag::warnings[]
+APOC contains over 450 functions and procedures.
```
Comment (JPryce-Aklundh) [inline readme.adoc:175]:
```suggestion
The APOC library contains over 450 functions and procedures.
```

File: readme.adoc [hunk 3]
```diff
@@ -172,13 +172,19 @@ You can find https://github.com/neo4j/apoc/releases/[all releases here].
...
+It is *not* recommended to load all of these into the dbms, but instead use the principle of least privilege.
+This principle dictates, that you should only load the functions and procedures that you strictly need to execute your queries.
+
+The procedures and functions that should be loaded can be specified using the Neo4j configuration setting `+dbms.security.procedures.allowlist+` in `$NEO4J_HOME/conf/neo4j.conf`.
+As an example, to load `apoc.math.maxInt` and all functions in the `apoc.rel` package, use `+dbms.security.procedures.allowlist=apoc.math.maxInt,apoc.rel.*+`.
```
Comment (JPryce-Aklundh) [inline readme.adoc:180]:
```suggestion
For example, to load `apoc.math.maxInt` and all functions in the `apoc.rel` package, use `+dbms.security.procedures.allowlist=apoc.math.maxInt,apoc.rel.*+`.
```

File: readme.adoc [hunk 4]
```diff
@@ -172,13 +172,19 @@ You can find https://github.com/neo4j/apoc/releases/[all releases here].
...
+
 For security reasons, procedures that use internal APIs are disabled by default.
-They can be enabled by specifying config in `$NEO4J_HOME/conf/neo4j.conf` e.g. `+dbms.security.procedures.unrestricted=apoc.*+`
+Also in this case, it is recommended to use the principle of least privilege and only unrestrict those procedures which you are certain to make use of.
```
Comment (JPryce-Aklundh) [inline readme.adoc:183]:
```suggestion
In this case, it is also recommended to use the principle of least privilege and only unrestrict those procedures which you are certain to use.
```

File: readme.adoc [hunk 5]
```diff
@@ -172,13 +172,19 @@ You can find https://github.com/neo4j/apoc/releases/[all releases here].
...
+The procedures can be unrestricted using the Neo4j configuration setting `+dbms.security.procedures.unrestricted+` in `$NEO4J_HOME/conf/neo4j.conf` e.g. `+dbms.security.procedures.unrestricted=apoc.cypher.runFirstColumn,apoc.cypher.doIt+`.
```
Comment (JPryce-Aklundh) [inline readme.adoc:184]:
```suggestion
Procedures can be unrestricted using the Neo4j configuration setting `+dbms.security.procedures.unrestricted+` in `$NEO4J_HOME/conf/neo4j.conf`. 
For example, to unrestrict `apoc.cypher.runFirstColumn` and `apoc.cypher.doIt`, use: `+dbms.security.procedures.unrestricted=apoc.cypher.runFirstColumn,apoc.cypher.doIt+`.
```

File: readme.adoc [hunk 6]
```diff
@@ -172,13 +172,19 @@ You can find https://github.com/neo4j/apoc/releases/[all releases here].
...
 
-If you want to do this when using the Neo4j Docker container, you need to amend `+-e NEO4J_dbms_security_procedures_unrestricted=apoc.\\\*+` to your `docker run ...` command.
+If you want to unrestrict a whole package of APOC functions and procedures when using the Neo4j Docker container, you need to amend e.g. `+-e NEO4J_dbms_security_procedures_unrestricted=apoc.cypher.\\\*+` to your `docker run ...` command.
```
Comment (JPryce-Aklundh) [inline readme.adoc:186]:
```suggestion
To unrestrict a whole package of APOC functions and procedures when using the Neo4j Docker container, you need to amend the Docker environment variables. 
For example, to add the full `apoc.cypher` package, add `+-e NEO4J_dbms_security_procedures_unrestricted=apoc.cypher.\\\*+` to your `docker run ...` command.
```

**Comment by AzuObs** [comment]
What do you think about the idea of keeping this section in the readme, but instead of going into great detail just giving a link pointing to the official documentation for this section? I will reach out to you on Slack or in another PR to give more context about why I think this might be a good idea.

**Comment by irene221b** [comment]
lgtm

**Comment by Lojjs** [comment]
> What do you think about the idea of keeping this section in the readme, but instead of going into great detail we could just link to the official documentation for this section? I will reach out to you on Slack or in another PR to give more context about why I think this might be a good idea.

That sounds like a good idea so we do not have to maintain duplicate documentation

File: readme.adoc [hunk 7]
```diff
@@ -172,13 +172,9 @@ You can find https://github.com/neo4j/apoc/releases/[all releases here].
...
-For security reasons, procedures that use internal APIs are disabled by default.
-They can be enabled by specifying config in `$NEO4J_HOME/conf/neo4j.conf` e.g. `+dbms.security.procedures.unrestricted=apoc.*+`
-
-If you want to do this when using the Neo4j Docker container, you need to amend `+-e NEO4J_dbms_security_procedures_unrestricted=apoc.\\\*+` to your `docker run ...` command.
-The three backslashes are necessary to prevent wildcard expansions.
-
-You _can_ also allow procedures and functions in general to be loaded using: `+dbms.security.procedures.allowlist=apoc.coll.*,apoc.load.*+`
+For security reasons, procedures and functions that use internal APIs are disabled by default.
+Loading and enablement of APOC procedures and functions can be configured using the Neo4j config file.
+For more details , see https://neo4j.com/docs/apoc/current/installation/#restricted[the APOC installation documentation].
```
Comment (JPryce-Aklundh) [inline readme.adoc:177]:
```suggestion
For more details, see https://neo4j.com/docs/apoc/current/installation/#restricted[the APOC installation documentation].
```

File: readme.adoc [hunk 8]
```diff
@@ -172,13 +172,9 @@ You can find https://github.com/neo4j/apoc/releases/[all releases here].
...
 // tag::warnings[]
-For security reasons, procedures that use internal APIs are disabled by default.
-They can be enabled by specifying config in `$NEO4J_HOME/conf/neo4j.conf` e.g. `+dbms.security.procedures.unrestricted=apoc.*+`
-
-If you want to do this when using the Neo4j Docker container, you need to amend `+-e NEO4J_dbms_security_procedures_unrestricted=apoc.\\\*+` to your `docker run ...` command.
-The three backslashes are necessary to prevent wildcard expansions.
-
-You _can_ also allow procedures and functions in general to be loaded using: `+dbms.security.procedures.allowlist=apoc.coll.*,apoc.load.*+`
+For security reasons, procedures and functions that use internal APIs are disabled by default.
+Loading and enablement of APOC procedures and functions can be configured using the Neo4j config file.
```
Comment (JPryce-Aklundh) [inline readme.adoc:176]:
```suggestion
Loading and enabling APOC procedures and functions can be configured using the Neo4j config file.
```

**Review by AzuObs — APPROVED** [review]
I agree with Jens' suggestions but assuming they are acted upon this gets +1 from me

**Pull request merged**